libs/wingc/src/jsify.rs

```diff
@@ -710,6 +695,315 @@ impl JSifier {
 	fn jsify_class_member(&self, member: &ClassMember) -> String {
 		format!("{};", self.jsify_symbol(&member.name))
 	}
+
+	fn jsify_resource(
+		&self,
+		env: &SymbolEnv,
+		name: &Symbol,
+		phase: Phase,
+		parent: &Option<Type>,
+		constructor: &Constructor,
+		methods: &[&(Symbol, FunctionDefinition)],
+	) -> String {
+		assert!(phase == Phase::Preflight);
+
+		// Lookup the resource type
+		let resource_type = env.lookup(name, None).unwrap().as_type().unwrap();
+
+		// Get fields to be captured by resource's client
+		let captured_fields = self.get_captures(resource_type);
+
+		// Jsify inflight client
+		let inflight_methods = methods
+			.iter()
+			.filter(|(_, m)| m.signature.flight == Phase::Inflight)
+			.map(|t| *t)
+			.collect::<Vec<_>>();
```

Comment:
Curious: do we really need this "collect"? Why?

Reply:
We use `inflight_methods` in two places so I figured we can collect this and use it twice. Not perfect but was quicker than passing an iterator to `jsify_resource_client` and then dealing with resetting / cloning the iter.